ckcp/gitops.sh

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env bash
+
+#quit if exit status of any cmd is a non-zero value
+set -euo pipefail
+# set -x
+
+SCRIPT_DIR="$(cd "$(dirname "$0")" >/dev/null ; pwd)"
+GITOPS_DIR="$(dirname "$SCRIPT_DIR")/gitops"
+KUBECONFIG_CLUSTER="${KUBECONFIG:-$HOME/.kube/config}"
+
+
+
+install_openshift_gitops(){
+  APP="openshift-gitops"
+  export KUBECONFIG="$KUBECONFIG_CLUSTER"
+
+  local ns="openshift-operators"
+
+  #############################################################################
+  # Install the gitops operator
+  #############################################################################
+  echo -n "  - OpenShift-GitOps: "
+  if ! oc get subscriptions -n "$ns" openshift-gitops-operator >/dev/null 2>&1; then
+    oc apply -f "$GITOPS_DIR/$APP.yaml" --wait >/dev/null
+  fi
+  echo "OK"
+
+  #############################################################################
+  # Wait for the URL to be available
+  #############################################################################
+  echo -n "  - ArgoCD dashboard: "
+  test_cmd="oc get route/openshift-gitops-server --ignore-not-found -n $APP -o jsonpath={.spec.host}"
+  ARGOCD_HOSTNAME="$(${test_cmd[@]})"
+  while ! curl --fail --insecure --output /dev/null --silent "https://$ARGOCD_HOSTNAME"; do
+    echo -n "."
+    sleep 2
+    ARGOCD_HOSTNAME="$(${test_cmd[@]})"
+  done
+  echo "OK"
+  echo "  - ArgoCD URL: https://$ARGOCD_HOSTNAME"
+
+  #############################################################################
+  # Post install
+  #############################################################################
+
+  # Allow ArgoCD to control resources outside of its namespace
+  oc adm policy add-cluster-role-to-user cluster-admin -z openshift-gitops-argocd-application-controller -n openshift-gitops >/dev/null
+
+  # Log into ArgoCD
+  echo -n "  - ArgoCD Login: "
+  local argocd_password
+  argocd_password="$(oc get secret openshift-gitops-cluster -n openshift-gitops -o jsonpath="{.data.admin\.password}" | base64 --decode --ignore-garbage)"
+  argocd login "$ARGOCD_HOSTNAME" --insecure --username admin --password "$argocd_password" >/dev/null
+  echo "OK"
```

Comment:
the output is confusing:
```
./ckcp/gitops.sh 
[openshift-gitops]
  - OpenShift-GitOps: OK
  - ArgoCD dashboard: OK
  - ArgoCD URL: https://openshift-gitops-server-openshift-gitops.apps.mkovarik.test.redhat.com
  - ArgoCD Login: WARN[0002] Failed to invoke grpc call. Use flag --grpc-web in grpc calls. To avoid this warning message, use flag --grpc-web. 
OK
```
Last message is OK but that Warning is not nice, can be removed by adding `--grpc-web` to argocd command.